Add unique ids, use them when adding/removing methods

portdeveloper opened the pull request:
## Description

This PR:

- adds unique ids to each method in an abi
- uses those unique ids to handle add and remove operations
- also pushes the unique ids to the url bar

Creating this PR as a draft because of the obvious messy code.

## ToDo - Possible Improvements
- [ ] Code cleanup
- [x] Better types?
- [ ] any of your suggestions!

## Before the changes


https://github.com/BuidlGuidl/abi.ninja/assets/108868128/a5717d66-ffe0-4fd0-9032-053ef926989a

## After the changes


https://github.com/BuidlGuidl/abi.ninja/assets/108868128/c828f95a-fc58-45c8-aa9f-fa7b64ce7426




## Additional Information

- [ ] I have read the [contributing docs](https://github.com/BuidlGuidl/abi.ninja/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [ ] This is not a duplicate of any [existing pull request](https://github.com/BuidlGuidl/abi.ninja/pulls)

## Related Issues

_Closes #66 

_Note: If your changes are small and straightforward, you may skip the creation of an issue beforehand and remove this section. However, for medium-to-large changes, it is recommended to have an open issue for discussion and approval prior to submitting a pull request._

Your ENS/address:

Please be nitpicky!!

technophile-04 commented:
Hey thanks @portdeveloper, this is great and makes sense ! 

Just found a bug : 


https://github.com/BuidlGuidl/abi.ninja/assets/80153681/5175b53d-0f74-4e76-ad97-6e2b0d9b1d5a

The Write methods are not highlighted when you select them in sidebar

--- 

Also could you please fix the lint error ? Maybe use `useMemo` for `readMethodsWithInputsAndWriteMethods `and add it as dependency to useEffect. 

--- 

In future maybe could find a way where we only append `functionName_uuid`  only when there is overload otherwise we just append functionName ( similar to how we have it now). We can tackle this in future no need to tackle it in this PR 🙌

portdeveloper commented:
Thanks for pointing that out @technophile-04 !
I paid attention to so many other things but haven't noticed that write methods selector was not working :D 
Fixed that also added a useMemo for readMethodsWithInputsAndWriteMethods.
Thanks!

technophile-04 reviewed (approved):
Tysm ! Looking great !!!
